source/simulator/core.cc

```diff
@@ -725,15 +738,25 @@ namespace aspect
                                                           std_cxx11::_1),
                                                           introspection.component_masks.temperature.first_selected_component(),
                                                           introspection.n_components),
-                                                      current_constraints,
+                                                      new_current_constraints,
                                                       introspection.component_masks.temperature);
           }
       }
 
+    if (!boundary_temperature_manager.allows_fixed_temperature_on_outflow_boundaries())
+      restore_outflow_boundary_ids(boundary_id_offset);
+
     // If there are fixed boundary compositions,
     // update the composition boundary condition.
     boundary_composition_manager.update();
 
+    // If we do not want to prescribe Dirichlet boundary conditions on outflow boundaries,
+    // we update the boundary indicators of all faces that belong to ouflow boundaries
+    // so that they are not in the list of fixed composition boundary indicators any more.
+    // We will undo this change in a later step, after the constraints have been set.
+    if (!boundary_composition_manager.allows_fixed_composition_on_outflow_boundaries())
```

Comment:
this is a copy of the comment from above except that you want to apply it for the compositional fields now. maybe say something along the lines "Use the same trick for marking up outflow boundary conditions for compositional fields as we did above already for the temperature."

I guess you convinced yourself that you need the correct boundary indicators in the call to `update` above in line 751 (new) and that you can't just avoid the restore/replace pair in line 747/758?

Reply:
No, I just thought it would be cleaner like this. There are different parameters in the if-statement, so we would have to do something like 
``
    if (!boundary_temperature_manager.allows_fixed_temperature_on_outflow_boundaries()
    	&& boundary_composition_manager.allows_fixed_composition_on_outflow_boundaries())
      restore_outflow_boundary_ids(boundary_id_offset);
    if (!boundary_composition_manager.allows_fixed_composition_on_outflow_boundaries()
        && boundary_temperature_manager.allows_fixed_temperature_on_outflow_boundaries())
      replace_outflow_boundary_ids(boundary_id_offset);
``

Reply:
Ah, I missed that. I thought you're just restoring it for the `update` call immediately above.

OK then, just edit the comment.